de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/model/interceptor/C_BPartner.java

```diff
@@ -60,4 +70,28 @@ public void createBPartnerStatsRecord(@NonNull final I_C_BPartner bpartner)
 	{
 		Services.get(IBPartnerStatsDAO.class).retrieveBPartnerStats(bpartner);
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_CHANGE}, ifColumnsChanged = {I_C_BPartner.COLUMNNAME_C_BP_Group_ID})
+	public void updateBPartnerStatsRecord(@NonNull final I_C_BPartner bpartner)
+	{
+		final IBPartnerStats bpartnerStats = Services.get(IBPartnerStatsDAO.class).retrieveBPartnerStats(bpartner);
+		final I_C_BP_Group bpGroup = bpartner.getC_BP_Group();
+		final String creditStatus = bpGroup.getSOCreditStatus();
+		if (!Check.isEmpty(creditStatus,true))
```

Comment:
don't we have to update BP's stats even when creditStatus is empty?

Reply:
also why not moving this code to BPartnerStatsBL?

Reply:
The credit status is updated when changes are made regarding partner credit limit or document.
If the credit status in bpartner group is empty, I don;t want to touch partner's credit status.

Reply:
but, yes, the logic we could have it in BPartnerStatsBL. Better place there